src/main/java/org/liquidengine/legui/marshal/json/JsonMarshaller.java

```diff
@@ -0,0 +1,291 @@
+package org.liquidengine.legui.marshal.json;
+
+import static org.liquidengine.legui.binding.model.BindingUtilities.getFieldValue;
+import static org.liquidengine.legui.marshal.json.JsonSerisalizeCreator.createAdapter;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonNull;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonPrimitive;
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import org.liquidengine.legui.binding.BindingRegistry;
+import org.liquidengine.legui.binding.converter.AbstractClassConverter;
+import org.liquidengine.legui.binding.model.AbstractClassBinding;
+import org.liquidengine.legui.binding.model.Binding;
+import org.liquidengine.legui.binding.model.BindingUtilities;
+
+/**
+ * Json marshaller.
+ *
+ * @author ShchAlexander.
+ */
+public final class JsonMarshaller {
+
+    public static final String TYPE_PROPERTY = "@type";
```

Comment:
One of the reasons for json/xml serialization is to change the gui without knowing the code, adding the classes in every JsonObject, it's a bit cumbersome, a solution could be adding alias to the classes so you put `"type": "Panel"` instead of `"@type":"org.liquidengine.legui.component.Panel"`

Reply:
Thougth about that, probably can use to value from class-binding element.

Reply:
Created next logic: If class is same as described in binding - representation will have type alias. In other case it will have class name as attribute.